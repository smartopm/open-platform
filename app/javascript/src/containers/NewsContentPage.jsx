import React from 'react'
import IframeContainer from '../components/IframeContainer'

export default function NewsContentPage() {
  return <IframeContainer link="https://doublegdp.gitlab.io/nkwashi-content/" />
}
