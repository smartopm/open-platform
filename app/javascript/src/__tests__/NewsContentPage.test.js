import React from 'react'
import { mount } from 'enzyme'
import NewsContentPage, { formatUrl } from '../containers/NewsContentPage'

describe('news content page', () => {
  const page = mount(<NewsContentPage />)
  it('renders and matches the snapshot', () => {
    expect(page).toMatchSnapshot()
  })
  it('should include dom elements when rendered', () => {
    expect(page.find('iframe')).toHaveLength(1)
    expect(page.find('div')).toHaveLength(1)
  })
})

describe('format url function', () => {
  // give a page like this https://olivier.dgdp.site/news/policies/gate-access-policy/
  // should return this policies/gate-access-policy/
  const currentPath =
    'https://olivier.dgdp.site/news/policies/gate-access-policy/'
  it('should extract the url params from the given path', () => {
    expect(formatUrl(currentPath, 'news')).toBe('policies/gate-access-policy/')
  })
  it('should extract the url params from other tags besides news', () => {
    expect(formatUrl(currentPath, 'policies')).toBe('gate-access-policy/')
  })
})
