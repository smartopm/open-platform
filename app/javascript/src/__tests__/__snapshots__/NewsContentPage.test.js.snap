// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`news content page renders and matches the snapshot 1`] = `ReactWrapper {}`;
